swift/ql/src/queries/Security/CWE-135/StringLengthConflation.ql

```diff
@@ -58,39 +58,41 @@ class StringLengthConflationConfiguration extends DataFlow::Configuration {
   override predicate isSource(DataFlow::Node node, string flowstate) {
     // result of a call to `String.count`
     exists(MemberRefExpr member |
-      member.getBase().getType().getName() = "String" and
+      member.getBase().getType().(NominalType).getABaseType*().getName() = "String" and
```

Comment:
Could you not use the predicate I added in https://github.com/github/codeql/pull/10254 and rewrite this to:
```ql
member.getBase().(ClassDecl).getABaseTypeDecl*().getName()
```
(or something along those lines)?

Reply:
`getBase()` returns an `Expr`, not a `ClassDecl`, so I think an extra step of some form is required either way.

Reply:
Doh, right 🤦 . Yeah, I see there are multiple ways of doing this, but none of them are _obviously_ shorter than the rest.